presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -666,12 +668,33 @@ output expressions:
 
     ORDER BY expression [ ASC | DESC ] [ NULLS { FIRST | LAST } ] [, ...]
 
-Each expression may be composed of output columns or it may be an ordinal
-number selecting an output column by position (starting at one). The
-``ORDER BY`` clause is evaluated after any ``GROUP BY`` or ``HAVING`` clause
+Each expression may be composed of output columns, or it may be an ordinal
+number selecting an output column by position, starting at one. The
+``ORDER BY`` clause is evaluated after any ``GROUP BY`` or ``HAVING`` clause,
 and before any ``OFFSET``, ``LIMIT`` or ``FETCH FIRST`` clause.
 The default null ordering is ``NULLS LAST``, regardless of the ordering direction.
 
+Note that, following the SQL specification, an ``ORDER BY`` clause only
+guarantees the order of rows for queries that immediately contain the clause.
+Presto follow that specification, and drops redundant usage of the clause to
+avoid negative performance impacts.
+
+In the following example, the clause only applies to the select statement.
+
+.. code-block:: none
+
+    INSERT INTO some_table
+    SELECT * FROM another_table
+    ORDER BY field
```

Comment:
wuold be good to have an example with inner query too,  like the second ex from the blog:

```
SELECT *
FROM some_table 
   JOIN (SELECT * FROM another_table ORDER BY field) u 
   ON some_table.key = u.key 
```

Reply:
@findepi : Currently, the insert statement mentioned above doesn't skip the order by in select - would be better if we have an insert which has an order in sub-query or use the query you have suggested above.

Reply:
@findepi : Currently, the insert statement mentioned above doesn't skip the order by in select - would be better if we have an insert which has an order in sub-query or use the query you have suggested above.

Reply:
> Currently, the insert statement mentioned above doesn't skip the order by in select -

@martint this (the insert statement mentioned above) is the first example from your blog post.
can you please verify it works as you intended it to?

Reply:
I added the second query as well, since it is a good example. 

I will need @findepi and @martint to confirm the behavior of Presto though.

Reply:
Insert should work as expected. I believe only “Create table as” has this issue due to how it manages scopes. Unless something else changed in how scopes are handled for Insert, too, and the functionality broke. I’ll verify in a bit.

Reply:
thx

Reply:
I have a feeling this is related ... https://github.com/prestosql/presto/pull/2044 @martint ?

Reply:
It is. With that fix, it should work for INSERT and CREATE TABLE AS. I confirmed it's currently not behaving as expected for INSERT, either.

Reply:
So what do we do with this documentation update and the blog post. Merge anyway and fix asap? I think we should since having those redundant order by clauses is a bad practice in any case .. no matter if Presto drops the sort or not.